module/VuFind/src/VuFind/Search/EDS/Options.php

```diff
@@ -115,28 +146,42 @@ class Options extends \VuFind\Search\Base\Options
      */
     protected $commonExpanders = [];
 
+    /**
+     * Search configuration
+     *
+     * @var \Laminas\Config\Config
+     */
+    protected $searchSettings;
+
     /**
      * Constructor
      *
      * @param \VuFind\Config\PluginManager $configLoader Configuration loader
-     * @param array                        $apiInfo      API information
+     * @param array|callable               $apiInfo      API information or callback
+     * to retrieve it
      */
     public function __construct(
         \VuFind\Config\PluginManager $configLoader,
         $apiInfo = null
     ) {
         $this->searchIni = $this->facetsIni = 'EDS';
-        $searchSettings = $configLoader->get($this->searchIni);
+        $this->searchSettings = $configLoader->get($this->searchIni);
         parent::__construct($configLoader);
         // 2015-06-30 RF - Changed to unlimited
         //$this->resultLimit = 100;
         $this->viewOptions = [
             'list|title' => 'Title View', 'list|brief' => 'Brief View',
             'list|detailed' => 'Detailed View'
         ];
-        $this->apiInfo = $apiInfo;
-        $this->setOptionsFromApi();
-        $this->setOptionsFromConfig($searchSettings);
```

Comment:
Is it possibly worth still running setOptionsFromConfig in the constructor? Or possibly splitting setOptionsFromConfig into two methods, one that overrides API-retrieved values and one that doesn't, so we can set up the non-API settings in the constructor and the rest as part of the API process? Otherwise, it seems that there might be some scenarios where we could construct an object and request a value and find that it hasn't been configured yet. (I think this is unlikely, but maybe not impossible).

What do you think?

Reply:
Yes, makes sense. I decided to change setOptionsFromConfig to only process options that are not dependent on API config and moved the rest to setOptionsFromApi. I also fixed invalid default for defaultView and removed extraneous parameters passing member variables around. I hope it makes more sense now!